CGE_equations.py

```diff
@@ -0,0 +1,688 @@
+        E (1D numpy array): Exports of good i
+        D (1D numpy array): Domestically produced good i
+
+    Returns:
+        Z (1D numpy array): Output from exporters CET production function
+    '''
+    Z = theta * (xie * E ** phi + xid * D ** phi) ** (1 / phi)
+    return Z
+
+
+def eqE(theta, xie, tauz, phi, pz, pe, Z):
+    '''
+    Supply of exports.
+
+    .. math::
+        E_{i} = \left(\theta_{i}^{\phi_{i}}\xi^{E}_{i}(1+\tau^{z}_{i}\frac{pz_{i}}{pe_{i}})\right)^{\frac{1}{1-\phi_{i}}}Z_{i}
+
+    Args:
+        theta (1D numpy array):
+        xie (1D numpy array): Share parameter for the share of exports of good i used by firms exporting good i
+        tauz (1D numpy array): Ad valorem tax rate on commodity i
+        phi (1D numpy array): Elasticity of substitution between exports (??) and domestically produced goods by firms exporting good i
+        pz (1D numpy array): price of output good i
+        pe (1D numpy array): Price of commodity i exports in domestic currency
+        Z (1D numpy array): Output from exporters CET production function
+
+    Returns:
+        E (1D numpy array): Exports of good i
+    '''
+    E = (theta ** phi * xie * (1 + tauz) * pz / pe) ** (1 / (1 - phi)) * Z
+    return E
+
+
+def eqD(theta, xid, tauz, phi, pz, pd, Z):
+    '''
+    Demand for domestic goods by exporters.
+
+    .. math::
+        D_{i} = \left(\theta_{i}^{\phi_{i}}\xi^{D}_{i}(1+\tau^{z}_{i}\frac{pz_{i}}{pd_{i}})\right)^{\frac{1}{1-\phi_{i}}}Z_{i}
+
+    Args:
+        theta (1D numpy array):
+        xid (1D numpy array): Share parameter for the share of domestically produced good i used by firms exporting good i
+        tauz (1D numpy array): Ad valorem tax rate on commodity i
+        phi (1D numpy array): Elasticity of substitution between exports (??) and domestically produced goods by firms exporting good i
+        pz (1D numpy array): price of output good i
+        pd (1D numpy array): price of domesically produced good i
+        Z (1D numpy array): Output from exporters CET production function
+
+    Returns:
+        D (1D numpy array): Demand for domestic good i by exporters.
+    '''
+    D = (theta ** phi * xid * (1 + tauz) * pz / pd) ** (1 / (1 - phi)) * Z
+    return D
+
+
+def eqpq(Q, Xp, Xg, Xv, X):
+    '''
+    Resource constraint.
+
+    .. math::
+        Q_{i} = X^{p}_{i} + X^{g}_{i} + X^{v}_{i} + \sum_{j}X_{i,j}
+    Args:
+        Q (1D numpy array): The domestic supply of good i, the Armington good
+        Xp (1D numpy array): Demand for production good i by consumers
+        Xg (1D numpy array): Government expenditures on commodity i
+        Xv (1D numpy array): Investment demand for each good i
+        X (2D numpy array): Demand for intermediate input i used in the
+            production of good j
+
+    Returns:
+        pq_error (1D numpy array): Error in resource constraint for each good i
+    '''
+    pq_error = Q - (Xp + Xg + Xv + X.sum(axis=1))
+    return pq_error
+
+
+def eqpf(F, Ff0):
```

Comment:
@DPrihardini Can you tell me what these last two equations represent?  I can see they have something to do with comparing quantities from the model to some in the data.

Reply:
eqpf is the market clearing condition for the labour market. Ff0[lab] is the labour supply from the data, but since we have a fixed labour supply in this model I'm comparing Ff0[lab] with the model's calculation of labour demand.

eqpk is the market clearing condition for capital, so same ideas as the labour market but capital Kk is endogenous in the model. I adjust for the fact that Ff[cap] is GOS whereas Kk is the physical stock.